### PR TITLE
Task/api 936 create practitionerrole resource

### PR DESCRIPTION
**Pull request by jjennings-va**
Added in practitioner role resource in accordance with the DSTU3 spec

http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-practitionerrole.html

https://vasdvp.atlassian.net/browse/API-936

### CODE REVIEW

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+  @Valid List<Reference> healthcareService;
+
+  @Valid List<PractitionerContactPoint> telecom;
+
+  @Valid List<Address> address;
+  @Valid Reference partOf;
+  @Valid List<OrganizationContact> contact;
+
+  PractitionerAvailableTime availableTime;
```
Comment (jgarnhart-va) [inline provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: Array, not single object.

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+  PractitionerNotAvailable notAvailable;
```
Comment (jgarnhart-va) [inline provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: Array, not single object.

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+  @Valid List<Reference> location;
+
+  @Valid List<Reference> healthcareService;
+
+  @Valid List<PractitionerContactPoint> telecom;
+
+  @Valid List<Address> address;
+  @Valid Reference partOf;
+  @Valid List<OrganizationContact> contact;
```
Comment (jgarnhart-va) [inline provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: I don't see `address`, `partOf`, or `contact` in the spec.

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java (hunk 2 above)
Comment (jjennings-va) [reply provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: fixed

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java (hunk 1 above)
Comment (jjennings-va) [reply provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: fixed, good catch

File: provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java (hunk 3 above)
Comment (jjennings-va) [reply provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/PractitionerRole.java]: copy paste error. Fixed

**Pull request merged**